### PR TITLE
[libbeat] Remove global loggers from libbbeat/dashboard

### PR DESCRIPTION
**Pull request by fearful-symmetry**
## What does this PR do?

See elastic/beats#18500 for the meta-issue. This remove any global loggers. I was a tad hesitant to turn "soft" errors into hard errors, so I worked around some of the error handling for the sake of not breaking anything. This still technically contains "breaking" changes, so not sure if we wanna backport it.

## Why is it important?

We want to remove the global logger from libbeat.

## Checklist

- [X] My code follows the style guidelines of this project
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

### CODE REVIEW

File: libbeat/dashboards/kibana_loader.go [hunk 1]
```diff
@@ -153,6 +155,7 @@ func (loader KibanaLoader) statusMsg(msg string, a ...interface{}) {
 	if loader.msgOutputter != nil {
 		loader.msgOutputter(msg, a...)
 	} else {
-		logp.Debug("dashboards", msg, a...)
+		logger := logp.NewLogger("dashboards")
```
Comment (urso) [inline libbeat/dashboards/kibana_loader.go:158]: maybe we want to create the logger in NewKibanaLoader?

File: libbeat/dashboards/kibana_loader.go [hunk 2]
```diff
@@ -153,6 +157,6 @@ func (loader KibanaLoader) statusMsg(msg string, a ...interface{}) {
 	if loader.msgOutputter != nil {
 		loader.msgOutputter(msg, a...)
 	} else {
-		logp.Debug("dashboards", msg, a...)
+		loader.defaultLogger.Debugf("%s %v", msg, a)
```
Comment (urso) [inline libbeat/dashboards/kibana_loader.go:160]: the old implementation looks like "statusMsg" implements a Printf like interface (better named would have been `statusMsgf`. 

`s/loader.defaultLogger.Debugf("%s %v", msg, a)/loader.defaultLogger.debugf(msg, a...)/`

Is MsgOutputer used somewhere? It kind of looks like a 'duplicate' interface for logging.

Comment (fearful-symmetry) [reply libbeat/dashboards/kibana_loader.go:160]: Yah, it was one of those things where whatever calls the loader also asks for that message handler option, so it seems to go fairly high up the call stack and I was hesitant to mess with it.

Comment (urso) [reply libbeat/dashboards/kibana_loader.go:160]: I see.

Let's change this line to this (to restore the behavior from before this PR):

```
loader.defaultLogger.Debugf(msg, a...)
```

**Pull request merged**